tests/unittest/aws_test.py

```diff
@@ -0,0 +1,379 @@
+
+    found_instances = FAKE_AWS_ACCOUNT.GetInstancesByNameOrId(
+        instance_name=FAKE_INSTANCE_WITH_NAME.name)
+    self.assertEqual(1, len(found_instances))
+    self.assertEqual(
+        'fake-instance-with-name-id', found_instances[0].instance_id)
+
+    self.assertRaises(ValueError, FAKE_AWS_ACCOUNT.GetInstancesByNameOrId)
+    self.assertRaises(
+        ValueError,
+        FAKE_AWS_ACCOUNT.GetInstancesByNameOrId,
+        instance_id=FAKE_INSTANCE.instance_id,
+        instance_name=FAKE_INSTANCE_WITH_NAME.name)
+
+  @mock.patch('libcloudforensics.aws.AWSAccount.ListVolumes')
+  def testGetVolumeById(self, mock_list_volumes):
+    """Test that a volume of an account can be found by its ID."""
+    mock_list_volumes.return_value = MOCK_LIST_VOLUMES
+    found_volume = FAKE_AWS_ACCOUNT.GetVolumeById(
+        FAKE_VOLUME.volume_id)
+    self.assertIsInstance(found_volume, aws.AWSVolume)
+    self.assertEqual('fake-volume-id', found_volume.volume_id)
+    self.assertEqual('fake-zone-2', found_volume.region)
+    self.assertEqual('fake-zone-2b', found_volume.availability_zone)
+
+  @mock.patch('libcloudforensics.aws.AWSAccount.ListVolumes')
+  def testGetVolumesByName(self, mock_list_volumes):
+    """Test that a volume of an account can be found by its name."""
+    mock_list_volumes.return_value = MOCK_LIST_VOLUMES
+    found_volumes = FAKE_AWS_ACCOUNT.GetVolumesByName(
+        FAKE_BOOT_VOLUME.name)
+    self.assertEqual(1, len(found_volumes))
+    self.assertEqual('fake-boot-volume-id', found_volumes[0].volume_id)
+    self.assertEqual('fake-zone-2', found_volumes[0].region)
+    self.assertEqual('fake-zone-2b', found_volumes[0].availability_zone)
+
+    found_volumes = FAKE_AWS_ACCOUNT.GetVolumesByName(
+        'non-existent-volume-name')
+    self.assertEqual(0, len(found_volumes))
+
+  @mock.patch('libcloudforensics.aws.AWSAccount.ListVolumes')
+  def testGetVolumesByNameOrId(self, mock_list_volumes):
+    """Test that a volume of an account can be found by its name or ID."""
+    mock_list_volumes.return_value = MOCK_LIST_VOLUMES
+    found_volumes = FAKE_AWS_ACCOUNT.GetVolumesByNameOrId(
+        volume_id=FAKE_VOLUME.volume_id)
+    self.assertEqual(1, len(found_volumes))
+    self.assertEqual('fake-volume-id', found_volumes[0].volume_id)
+
+    found_volumes = FAKE_AWS_ACCOUNT.GetVolumesByNameOrId(
+        volume_name=FAKE_BOOT_VOLUME.name)
+    self.assertEqual(1, len(found_volumes))
+    self.assertEqual(
+        'fake-boot-volume-id', found_volumes[0].volume_id)
+
+    self.assertRaises(ValueError, FAKE_AWS_ACCOUNT.GetVolumesByNameOrId)
+    self.assertRaises(
+        ValueError,
+        FAKE_AWS_ACCOUNT.GetVolumesByNameOrId,
+        volume_id=FAKE_VOLUME.volume_id,
+        volume_name=FAKE_BOOT_VOLUME.name)
+
+  @mock.patch('libcloudforensics.aws.AWSAccount.ClientApi')
+  def testCreateDiskFromSnapshot(self, mock_ec2_api):
+    """Test the creation of a volume from a snapshot."""
```

Comment:
If you're testing the creation of a Volume, should the test function be renamed for consistency?